ads/google/a4a/docs/RTB-Implementation.md

```diff
@@ -0,0 +1,82 @@
+ 
+### RTB Bid Request
+ 
+Exchanges will need to indicate in the RTB bid request whether a page is built in AMP HTML, and any specific requirements or treatment of AMP Ads.  As of [OpenRTB 2.5](http://www.iab.com/wp-content/uploads/2016/03/OpenRTB-API-Specification-Version-2-5-FINAL.pdf), this is not yet included in the spec, but the proposed implementation to the IAB committee is as follows.
+ 
+**`Site` Object additional field: `amp`**
+
+A new field is added to the `Site` object of the OpenRTB standard to indicate whether a webpage is built on AMP HTML.  In OpenRTB 2.5, this is section 3.2.13.
+ 
+| Field         | Scope     | Type      | Default       | Description           |
+| ------------- |------     |-----      |:-------------:|-------------          |
+| `amp`         | optional  | integer   | -             | Whether the request is for an Accelerated Mobile Page. 0 = page is non-AMP, 1 = page is built with AMP HTML.  AMP status unknown if omitted. |
+ 
+**`Imp` Object additional field: `ampad`**
+
+A new field is added to the `Imp` object of the OpenRTB standard to provide more detail around AMP ad requirements and how AMP ads will load.  In OpenRTB 2.5, this is section 3.2.4.
+ 
+| Field         | Scope     | Type      | Default       | Description           |
+| ------------- |------     |-----      |:-------------:|-------------          |
+| `ampad`       | optional  | integer   | 1             | AMP Ad requirements and rendering behavior.  See AMP Ad Status table. |
+ 
+**AMP Ad Status Table**
+ 
+| Value        | Description            | 
+| ------------- |-------------          |
+| 1         | AMP Ad requirements are unknown.|
+| 2         | AMP Ads are not allowed.                |  
+| 3         | Either AMP Ads or non-AMP Ads are allowed; AMP Ads are not early rendered. | 
+| 4         | Either AMP Ads or non-AMP Ads are allowed, and AMP Ads are early rendered.|
+| 5         | AMP Ads are required.  Ads that are non-AMP may be rejected by the publisher.|
+| 500+      | Exchange-specific values; should be communicated to bidders *a priori*         |
+ 
+### RTB Bid Response
+ 
+SSPs will need to provide a new field in the bid response to allow bidders to return AMP Ad content, and RTB bidders will need to populate that field in order to return AMP Ads.  As of [OpenRTB 2.5](http://www.iab.com/wp-content/uploads/2016/03/OpenRTB-API-Specification-Version-2-5-FINAL.pdf), this is not yet included in the spec, but the proposed workflow is a new field that accepts a URL pointing to AMP ad content.  
+ 
+**`Bid` Object additional field: `ampadurl`**
+ 
+| Field         | Type     | Description        | 
+| ------------- |------     |-----              |
+| `ampadurl`       | string  | Optional means of conveying Amp Ad markup in case the bid wins; only one of `ampadurl` or `adm` should be set. Substitution macros (Section 4.4) may be included.  URL should point to a creative server containing valid AMP Ad html.           |  
```

Comment:
`adm` is thee standardized field for returning ad markup in openRTB (non-AMP specific).  See page 29 in this doc: http://www.iab.com/wp-content/uploads/2016/03/OpenRTB-API-Specification-Version-2-5-FINAL.pdf -- in the Bid object.